codegen-renderers/src/main/kotlin/io/vrap/codegen/languages/typescript/client/TypescriptMethodRenderer.kt

```diff
@@ -0,0 +1,134 @@
+            |   <${type.queryParams()}>
+            |
+            |   <${type.body()}>
+            |
+            |}
+            |
+            """.trimMargin().keepIndentation()
+
+        return TemplateFile(
+                relativePath = "$moduleName.ts",
+                content = content
+        )
+    }
+
+    fun Method.imports(moduleName: String): String {
+
+        val importTypes = mutableListOf(this.returnType())
+        importTypes.addAll(this.bodies.map { it.type })
+        return importTypes
+                .filter { it != null }
+                .map { it.toVrapType() }
+                .filter { it is VrapObjectType  }
+                .map { it as VrapObjectType }
+                .map { "import { ${it.simpleTSName()} } from '${relativizePaths(moduleName, it.`package`.replace(".","/"))}'" }
+                .joinToString(separator = "\n")
+
+    }
+
+    fun Method.body():String{
+        return if (this.hasNonEmptyBody()) {
+            "payload: ${this.bodyType()}"
+        } else {
+            ""
+        }
+    }
+
+    fun Method.bodyType():String {
+        return if (this.hasNonEmptyBody()) {
+            return this.bodies
+                    .map { it.type }
+                    .filter { it != null }
+                    .map { it.toVrapType().simpleTSName() }
+                    .joinToString(separator = " | ")
+
+        } else {
+            "void"
+        }
+    }
+
+    fun Method.hasNonEmptyBody() = this.bodies != null
+            && this.bodies.isNotEmpty()
+            && this.bodies.map { it.type != null }.reduce{acc, b ->  acc && b}
+
+    fun Method.pathVariables(): String {
+        val pathArguments = this.pathArguments()
+        if (pathArguments.isEmpty())
+            return ""
+        return """|
+            |pathVariables: {
+            |   <${pathArguments.map { "$it: string" }.joinToString(separator = ",\n")}>
+            |}
+        """.trimMargin()
+    }
+
+    fun Method.queryParams(): String {
+        val queryParams = this.queryParametersList()
+        if (queryParams.isEmpty())
+            return ""
+        return """|
+            |queryParams: {
+            |   <${this.queryParameters.map {"${it.name.tsRemoveRegexp()}${if (!it.required) "?" else ""}: ${it.type.toVrapType().simpleTSName()}" }.joinToString(separator = ",\n")}>
+            |}
+        """.trimMargin()
+    }
+
+
+
+    private fun relativizePaths(currentModule:String, targetModule : String) : String {
+        val currentRelative : Path = Paths.get(currentModule)
+        val targetRelative : Path = Paths.get(targetModule)
+        return currentRelative.relativize(targetRelative).toString().replaceFirst("../","")
+    }
+
+    private fun Method.pathArguments(): List<String> {
+        val urlPathParts = this.resource().fullUri.template.split("/").filter { it.isNotEmpty() }
+        return urlPathParts.filter { it.startsWith("{") && it.endsWith("}") }.map { it.replace("{", "").replace("}", "") }
```

Comment:
You could try to use the available methods on `fullUri` instead of your own implementation here 😉

Reply:
oh didn't know about that ⭐️ 
that would have saved me a bit of a hustle 😅